### PR TITLE
Fix #91: Add labels/badges on issues.

### LINKED ISSUE #91: Have badges for different types of bugs you've found
- See the error types on the homepage, there are 6

### PR DESCRIPTION
**Pull request by jajodiaraghav**
Fix #91

### CODE REVIEW

File: website/models.py [hunk 1]
```diff
@@ -120,10 +120,20 @@ def validate_image(fieldfile_obj):
...
 
 class Issue(models.Model):
+    labels = (
+        (1, 'Number Error'),
+        (2, 'Functional'),
+        (3, 'Performance'),
+        (4, 'Security'),
+        (5, 'Typo'),
+        (6, 'Design'),
+        (0, 'Unspecified')
```
Comment (souravbadami) [inline website/models.py]: How about changing *Unspecified* to *General* ? I think, that will cover some of the bugs which doesn't comes under the predefined category.

File: website/templates/base.html [hunk 2]
```diff
@@ -122,6 +122,15 @@
...
                     </div>
+                    <select name="label" class="form-control">
+                        <option value="0" selected="selected">Bug Type</option>
+                        <option value="1">Number error</option>
+                        <option value="2">Functional</option>
+                        <option value="3">Performance</option>
+                        <option value="4">Security</option>
+                        <option value="5">Typo</option>
+                        <option value="6">Design</option>
+                    </select>
```
Comment (souravbadami) [inline website/templates/base.html]: This should be rendered from the backend instead of hard coding the properties here. It'll guarantee a safe submission of our attribute integrity on both the sides.

File: website/templates/issue_edit.html [hunk 3]
```diff
@@ -64,6 +67,16 @@
...
+                <select name="label" class="form-control edit-label">
+                    <option value="0" selected="selected">Bug Type</option>
+                    <option value="1">Number error</option>
+                    <option value="2">Functional</option>
+                    <option value="3">Performance</option>
+                    <option value="4">Security</option>
+                    <option value="5">Typo</option>
+                    <option value="6">Design</option>
+                </select>
+
```
Comment (souravbadami) [inline website/templates/issue_edit.html]: Ditto as the earlier comment.

File: website/forms.py [hunk 4]
```diff
@@ -6,7 +6,7 @@ class IssueEditForm(forms.ModelForm):
 
     class Meta:
         model = Issue
-        fields = ['description', 'screenshot']
+        fields = ['description', 'screenshot', 'label']
```
Comment (souravbadami) [inline website/forms.py]: It is a required element.

**Review by souravbadami — CHANGES_REQUESTED** [review]
Hi @jajodiaraghav -- Looks great! I have left a few comments. Thanks!

File: website/forms.py (hunk 4 above)
Comment (jajodiaraghav) [inline website/forms.py]: I didn't get this part. Could be a bit more explicit?

Comment (souravbadami) [inline website/forms.py]: We can set the attributes of the form in the backend. So, if we declare a *required* clause within the form definition in the backend -- It'll make sure that it is required when it is rendered and thus form submission is declined if it is not filled. It also solves the problem that you asked on *Slack*.
https://docs.djangoproject.com/en/1.11/ref/forms/fields/#django.forms.Field.required

File: website/templates/base.html (hunk 2 above)
Comment (jajodiaraghav) [inline website/templates/base.html]: Although I have rendered the form element in the edit page from the backend, I couldn't do it in base.html. I am new to Django and would love to see how to do this. Could you please change this part on top of my commit?

Comment (souravbadami) [inline website/templates/base.html]: @jajodiaraghav -- What issue are you facing while using it in base.html ?

Comment (jajodiaraghav) [inline website/templates/base.html]: {{form.label}} does not render anything in base.html. I did it in a way similar to the one in issue_edit.html. It took me a lot of time trying to debug, but all in vain. Can we merge this and fix the base.html in a different commit. I want to work on issue_edit.html page but couldn't, as it would result in conflicts unless this gets merged.

Or else, please help me fix this.

Thanks :)

Comment (souravbadami) [inline website/templates/base.html]: Have a look at `website/forms.py` (IssueEditForm). The current implementation of the bug submission form needs a complete redesign. Maybe, we can do it completely on another PR.

**Review by souravbadami — APPROVED** [review]
